### ISSUE TITLE
Transfer Repo Ownership SSH Clone Redirect

### ISSUE THREAD

**Issue by NikolaDimic** [go-gitea/gitea#30565]
### Description

When transferring ownership of a repository (moving it to a new Org):
-  cloning the repository with the old URL using HTTP successfully makes a redirect to a new path (and pulls the repo) 
- cloning the repository with the old URL using SSH does not do a redirect.

Not sure if this should be a bug or a feature request, but seeing as redirects exist, and are kept in the Database I would expect it to work for both protocols.

### Gitea Version

1.21.1+21

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

I am hosting a standalone Gitea instance running on Ubuntu 20.04 as a systemd service.
Gitea binary: 1.21.1+21-gb7e3adc66

### Database

PostgreSQL

**Comment by lunny** [comment]
Maybe the SSH result is right because the transferring is not end.

**Comment by NikolaDimic** [comment]
The seps i used to reproduce this:
1. Have an Org with a repository:
<img width="936" alt="image" src="https://github.com/go-gitea/gitea/assets/43245519/e766916c-8745-419a-bb7f-c28142a9658a">
2. Transfer repository ownership to a new Org:
<img width="726" alt="image" src="https://github.com/go-gitea/gitea/assets/43245519/56ba5190-5ede-4134-b9be-64f299bd4242">
<img width="919" alt="image" src="https://github.com/go-gitea/gitea/assets/43245519/bd298171-55ac-46ac-bd1e-35f78d13ff13">

3. Try to clone with HTTP command with old path (✅ )
4. Try to clone with SSH command with old path (❌ )

<img width="963" alt="image" src="https://github.com/go-gitea/gitea/assets/43245519/94c52342-e535-4759-9bcf-8c77182600e8">


I believe the transfer is complete as the files are relocated on the server:
```
root@gitea-playground:~# ll /var/lib/gitea/data/gitea-repositories/old-owner/
drwxr-xr-x   2 git git 4096 Apr 18 13:24 ./
drwxr-xr-x 169 git git 4096 Apr 18 13:24 ../

root@gitea-playground:~# ll /var/lib/gitea/data/gitea-repositories/new-owner/
drwxr-xr-x   3 git git 4096 Apr 18 13:24 ./
drwxr-xr-x 169 git git 4096 Apr 18 13:24 ../
drwxr-xr-x   8 git git 4096 Apr 18 13:27 repo-transfer-owner.git/
```

and the redirect is added to Database:

```
gitea_db=# select * from repo_redirect where lower_name='repo-transfer-owner';
 id  | owner_id |     lower_name      | redirect_repo_id
-----+----------+---------------------+------------------
 147 |      472 | repo-transfer-owner |             1632
(1 row)
```

**Comment by koalajoe23** [comment]
Can confirm this behavior, running Gitea 1.24.5: The old HTTP Git-URL is properly redirected, SSH clone will fail.

Would be a really nice feature as communicating to everyone and changing repo location everywhere is a pain -_-

**Comment by hramrach** [comment]
Sounds like this would be a pain to implement.

Normally git ssh uses shell access to run some remote commands to work with the git repository.

The environment could be altered so that gitea is called in place of the usual git commands to do some magic on top. This does not seem to be something gitea would currently do.

Symlinks could be used with the standard git commands, and have the old location point to the current data. Something would need to maintain those over time, though. If there is a database of redirects these could be added to the filesystem as well with some hook but the syncing over time sounds awkward.

**Comment by hramrach** [comment]
Actually all SSH URLs are gitea@ so it must be using its own ssh implementation, and can do the magic there as well.

**Comment by koalajoe23** [comment]
If I understand the code correctly, The SSH daemon forces "gitea serv" as SSH command and gitea parses the supplied git url. I think we may do a redirect lookup as it's done for HTTP-Requests here: https://github.com/go-gitea/gitea/blob/07347634aa70defb7820332f4c860bb24f4ada2f/cmd/serv.go#L229-L230C2

Or am I mistaken?

**Comment by hramrach** [comment]
Right, it already uses a magic command to implement the SSH protocol. Adding more magic there should be feasible.

**Comment by koalajoe23** [comment]
Took a shot at it, see #35416
